code/game/machinery/doors/firedoor.dm

```diff
@@ -60,187 +40,100 @@
+				return
+			playsound(get_turf(src), 'sound/items/Ratchet.ogg', 50, 1)
+			user.visible_message("<span class='notice'>[user] starts undoing [src]'s bolts...</span>", \
+								 "<span class='notice'>You start unfastening [src]'s floor bolts...</span>")
+			if(!do_after(user, 50, target = src))
+				return
+			playsound(get_turf(src), 'sound/items/Deconstruct.ogg', 50, 1)
+			user.visible_message("<span class='notice'>[user] unfastens [src]'s bolts.</span>", \
+								 "<span class='notice'>You undo [src]'s floor bolts.</span>")
+			deconstruct(TRUE)
+			return
+		else if(istype(C, /obj/item/weapon/screwdriver))
+			user.visible_message("<span class='notice'>[user] [boltslocked ? "unlocks" : "locks"] [src]'s bolts...</span>", \
+								 "<span class='notice'>You [boltslocked ? "unlock" : "lock"] [src]'s floor bolts...</span>")
+			playsound(get_turf(src), 'sound/items/Screwdriver.ogg', 50, 1)
+			boltslocked = !boltslocked
 			return
-		if(blocked)
-			user.visible_message("\red \The [user] pries at \the [src] with \a [C], but \the [src] is welded in place!",\
-			"You try to pry \the [src] [density ? "open" : "closed"], but it is welded in place!",\
-			"You hear someone struggle and metal straining.")
-
-		if(stat & (BROKEN|NOPOWER) || !density || !alarmed)
-			user.visible_message("\red \The [user] forces \the [src] [density ? "open" : "closed"] with \a [C]!",\
-			"You force \the [src] [density ? "open" : "closed"] with \the [C]!",\
-			"You hear metal strain, and a door [density ? "open" : "close"].")
-
-		else if(allowed(user))
-			user.visible_message("\blue \The [user] lifts \the [src] with \a [C].",\
-			"\The [src] scans your ID, and obediently opens as you apply your [C].",\
-			"You hear metal move, and a door [density ? "open" : "close"].")
-
-		if(density)
-			spawn(0)
-				open()
-		else
-			spawn(0)
-				close()
-		return
-
-	var/access_granted = 0
-	var/users_name
-	if(!istype(C, /obj)) //If someone hit it with their hand.  We need to see if they are allowed.
-		if(allowed(user))
-			access_granted = 1
-		if(ishuman(user))
-			users_name = FindNameFromID(user)
 		else
-			users_name = "Unknown"
-
-	if(ishuman(user) && !stat && (istype(C, /obj/item/weapon/card/id) || istype(C, /obj/item/device/pda)))
-		var/obj/item/weapon/card/id/ID = C
-
-		if( istype(C, /obj/item/device/pda) )
-			var/obj/item/device/pda/pda = C
-			ID = pda.id
-		if(!istype(ID))
-			ID = null
-
-		if(ID)
-			users_name = ID.registered_name
-
-		if(check_access(ID))
-			access_granted = 1
-
-	var/answer = alert(user, "Are you sure you want to [density ? "open" : "close"] \the [src]?","\The [src] confirmation","Yes","No")
-	if(answer == "No")
-		return
-
-	if(user.stat || user.stunned || user.weakened || user.paralysis || get_dist(src, user) > 1)
-		to_chat(user, "Sorry, you must remain able bodied and close to \the [src] in order to use it.")
-		return
-
-	if(alarmed && density && !access_granted)

```

Comment:
I'd say we might want to keep this code in case we want to re-limit firedoor access (the legends say ponies removed it to help people escape the horrible monster ZAS), but we can simply recode it if we do so